sycl/unittests/Extensions/DeviceArchitectureOneArchSelected.cpp

```diff
@@ -0,0 +1,41 @@
+//==--------------- DeviceArchitectureOneArchSelected.cpp ------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <gtest/gtest.h>
+
+// define one of __SYCL_TARGET_INTEL_*** macro, e.g., the one for SKL
+#define __SYCL_TARGET_INTEL_GPU_SKL__ 1
+
+#include <sycl/ext/intel/experimental/device_architecture.hpp>
+
+using namespace sycl;
+using namespace sycl::detail;
+using namespace sycl::ext::intel::experimental;
+
+TEST(DeviceArchitectureTest, DeviceArchitectureOneArchSelected_If) {
+  bool res = false;
+  if_architecture_is<architecture::intel_gpu_skl>([&]() { res = true; });
+  ASSERT_TRUE(res);
+}
```

Comment:
I'd also add a simple negative case here:

```
   res = false;
   if_arc_is<wronge_arch>....
   ASSERT_FALSE(res);
```

It's kind of similar to the next case but is simpler so it would beneficial (IMO) to have it as well.